db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLDecimal.java

```diff
@@ -393,12 +393,14 @@ private boolean evaluateNull()
 	 */
 	public void writeExternal(ObjectOutput out) throws IOException {
         out.writeBoolean(isNull);
-        if (isNull)
-            return;
+		out.writeByte(precision);
```

Comment:
Sorry, I meant writing them unconditionally when the value isNull, basically move these 2 lines inside the isNull if.